src/main/scala/xiangshan/mem/sbuffer/Sbuffer.scala

```diff
@@ -984,6 +962,38 @@ class Sbuffer(implicit p: Parameters)
         difftestCommon.data   := wdata
         difftestCommon.mask   := wmask
 
+      } .otherwise {
+        difftestCommon.coreid := 0.U
```

Comment:
I'm considering that we may pass `dontCare=true` to DifftestModule to set default values.

Currently it sets DontCare for all fields and may bring functional issues. However, if we explicitly default the valid to false, then it should work perfectly. Then the code can be changed to

```scala
val difftest = DifftestModule(new DiffStoreEvent, delay = 2, dontCare = true)
when (isWline) {
  ...
}
```

Reply:
good idea

Reply:
I've updated difftest with the PR https://github.com/OpenXiangShan/difftest/pull/448. I think with this change, we can remove the `otherwise difftest.xxx := 0.U`. 

However, it may still bring errors if we use `when (cond) { val difftest = DifftestModule(...) }` (as in the previous version `elsewhen ... { val diff = DifftestModule(...) }`. There are some severe Chisel issues if we instantiate modules/regs/wires in `when`.